autonat.go

```diff
@@ -364,23 +447,41 @@ func (as *AmbientAutoNAT) tryProbe(p peer.ID) bool {
 }
 
 func (as *AmbientAutoNAT) probe(pi *peer.AddrInfo) {
```

Comment:
as this manager gets more involved, it might be worth splitting the probing / dialing logic (which partially resides in the ephemeral `Client`) into one file (e.g. "natprobingmanager"), and the tracking event queue ("natstatusmanager") into a different one?